qadi_simple.py

```
@@ -55,6 +57,86 @@
         self.prompts = SimplerQADIPrompts()
 
 
+def extract_key_solutions(hypotheses: List[str], final_answer: str, action_plan: List[str]) -> List[str]:
```

Ruff (0.12.2)

`extract_key_solutions` is too complex (16 > 10)

(C901)